components/service/glean/src/main/java/mozilla/components/service/glean/ping/BaselinePing.kt

```diff
@@ -69,4 +72,45 @@ internal class BaselinePing(applicationContext: Context) {
             it.id
         }
     }
+
+    /**
+     * Gets a gecko-compatible locale string (e.g. "es-ES" instead of Java [Locale]
+     * "es_ES") for the default locale.
+     *
+     * This method approximates the API21 method [Locale.toLanguageTag].
+     *
+     * @return a locale string that supports custom injected languages.
+     */
+    internal fun getLanguageTag(): String {
+        // Thanks to toLanguageTag() being introduced in API21, we could have
+        // simple returned `locale.toLanguageTag();` from this function. However
+        // what kind of languages the Android build support is up to the manufacturer
+        // and our apps usually support translations for more rare languages, through
```

Comment:
Does this imply that the browser (or android-components) allows selection of a locale that isn't available on the system, or isn't the same as what the user has selected system-wide?  Just want to clarify because the code here is using the system's default (which I assume is what the user selected in the system-wide settings) and not some android-components-specific API.

Reply:
As far as I understood from @pocmo , a-c can inject rare languages with its own language picker, that are then picked up by this. @pocmo is also reviewing this to make sure I didn't misunderstand :)

Reply:
Yeah, correct. In order to support languages that are not installed on the system (our apps have a custom language setting) we inject a different Locale at runtime. That's the code in Focus:
https://github.com/mozilla-mobile/focus-android/blob/master/app/src/main/java/org/mozilla/focus/locale/LocaleManager.java

We _could_ test and verify how `toLanguageTag()` behaves with that and then just use it here. But it may be hard to make an assumption over all variations of Android we may see.